src/hw_tools.py

```diff
@@ -229,6 +237,86 @@ def __init__(self, channel: str) -> None:
+        """Install the driver and NVIDIA utils."""
+        self._install_nvidia_drivers()
+        self._install_nvidia_utils()
+
+    def _install_nvidia_drivers(self) -> None:
+        """Install the NVIDIA driver if not present."""
+        if Path("/proc/driver/nvidia/version").exists():
+            logger.info("NVIDIA driver already installed in the machine")
+            return
+
+        logger.info("Installing NVIDIA driver")
+        apt.add_package("ubuntu-drivers-common", update_cache=True)
+
+        # output what driver was installed helps gets the version installed later
+        cmd = (
+            "ubuntu-drivers install --gpgpu --package-list "
+            f"{self.snap_common}/nvidia-installed-pkgs.txt"
+        )
+        try:
+            # This can be changed to check_call and not rely in the output if this is fixed
+            # https://github.com/canonical/ubuntu-drivers-common/issues/106
+            result = subprocess.check_output(cmd.split(), text=True)
+
+        except subprocess.CalledProcessError as err:
+            logger.error("Failed to install the NVIDIA driver: %s", err)
+            raise err
+
+        if "No drivers found for installation" in result:
+            logger.warning(
+                "No drivers for the NVIDIA GPU were found. Manual installation is necessary"
+            )
+            raise ResourceInstallationError(self._name)
+
+        logger.info("NVIDIA driver installed")
+
+    def _install_nvidia_utils(self) -> None:
+        """Install the nvidia utils to be able to use nvidia-smi."""
+        nvidia_pkg = Path(self.snap_common / "nvidia-installed-pkgs.txt")
+        if not nvidia_pkg.exists():
+            logger.debug("nvidia-utils not installed by the charm")
+            return
+
+        installed_pkg = nvidia_pkg.read_text(encoding="utf-8").splitlines()[0]
+        logger.debug("installed driver from hardware-observer: %s", installed_pkg)
+        nvidia_version = installed_pkg.split("-")[-2]
+
+        if not nvidia_version.isdigit():
+            logger.warning(
+                "driver %s is an unexpected format and nvidia-utils was not installed",
+                installed_pkg,
+            )
+            return
+
+        pkg = f"nvidia-utils-{nvidia_version}-server"
+        apt.add_package(pkg, update_cache=True)
+        logger.info("installed %s", pkg)
+        return
+
+    def check(self) -> bool:
+        """Check if nvidia-smi is working."""
+        try:
+            subprocess.check_call("nvidia-smi", timeout=60)
+            return True
+        except (FileNotFoundError, subprocess.CalledProcessError) as e:
+            logger.error(e)
+            logger.warning(
+                "nvidia-smi is not working. It's necessary to manually remove and install "
+                "a different NVIDIA driver until nvidia-smi is working. See the docs for more "
+                "details: https://ubuntu.com/server/docs/nvidia-drivers-installation"
```

Comment:
well, can we be confident the installed driver will need to be removed? I would suggest using a more vague statement
```suggestion
                "nvidia-smi is not working. Ensure the correct driver is installed. See the docs for more "
                "details: https://ubuntu.com/server/docs/nvidia-drivers-installation"
```